Classes/Services/MetsExporter.php

```diff
@@ -316,6 +316,28 @@ public function parseXPath($xPath)
         return $xml;
     }
 
+    public function sortPredicates($xpath) {
+        $search = '/\[(.*)\]/U';
+
+        preg_match_all($search, $xpath, $matches);
+
+        $last = '';
+        $predicateString = '';
+
+        foreach ($matches[1] as $key => $value) {
+            if (substr($value, 0, 1) == '@') {
+                $predicateString .= '[' . $value . ']';
+            } else {
+                $last .= '[' . $value . ']';
+            }
+        }
+
+        $xpathExploded = explode("[", $xpath);
+        $newXpath = $xpathExploded[0] . $predicateString . $last;
```

Comment:
If `$xpath` contains no conditions (no brackets) explode() returns an empty array and accessing $xpathExploded[0] results in an out-of-bound array access. If the `preg_match_all` doesn't match anything, no XPath conditions need to be "sorted".